include/dxc/WinAdapter.h

```diff
@@ -912,16 +912,60 @@ unsigned int SysStringLen(const BSTR bstrString);
 #define CP_ACP 0
 #define CP_UTF8 65001 // UTF-8 translation.
 
-// Convert Windows codepage value to locale string
-const char *CPToLocale(uint32_t CodePage);
+// RAII style mechanism for setting/unsetting a locale for the specified Windows
+// codepage
+class ScopedLocale {
+  const char *m_locale;
+
+public:
+  explicit ScopedLocale(uint32_t codePage)
+      : m_locale(SetLocaleForCodePage(LC_ALL, codePage)) {}
+  ~ScopedLocale() {
+    if (m_locale != nullptr) {
+      setlocale(LC_ALL, m_locale);
+    }
+  }
+  bool IsSupported() { return (m_locale != nullptr); }
+
+private:
+  // Sets a locale for the specified Windows codepage
+  const char *SetLocaleForCodePage(int Category, uint32_t CodePage) {
+#ifdef __APPLE__
+    switch (CodePage) {
+    case CP_ACP:
+      return setlocale(Category, "en_US.ISO8859-1");
```

Comment:
Existing, but isn't CP_ACP supposed to refer to the system default ANSI code page, rather than this specific code page?  For the compiler, and on Apple/Linux platforms, shouldn't that be considered whatever's already set in locale?

Reply:
It turns out that CP_ACP is _never_ used with this code!